gdspy/library.py

```diff
@@ -577,16 +577,15 @@ def get_bounding_box(self):
             for reference in self.references:
                 reference_bb = reference.get_bounding_box()
                 if reference_bb is not None:
-                    bb[0, 0] = min(bb[0, 0], reference_bb[0, 0])
-                    bb[0, 1] = min(bb[0, 1], reference_bb[0, 1])
-                    bb[1, 0] = max(bb[1, 0], reference_bb[1, 0])
-                    bb[1, 1] = max(bb[1, 1], reference_bb[1, 1])
+                    all_polygons.extend([reference_bb])
             if len(all_polygons) > 0:
                 all_points = numpy.concatenate(all_polygons).transpose()
                 bb[0, 0] = min(bb[0, 0], all_points[0].min())
                 bb[0, 1] = min(bb[0, 1], all_points[1].min())
                 bb[1, 0] = max(bb[1, 0], all_points[0].max())
                 bb[1, 1] = max(bb[1, 1], all_points[1].max())
+            else:
+                return None
```

Comment:
I believe this should set `bb = None` instead of directly returning None. That way, the cell bounding box will be correctly cached.

Reply:
we should do this for the top if statement too then, yes?

Reply:
The top condition is just a quick short-circuit for the obvious case. It doesn't even checks `_bb_valid`, so it would actually make the check longer. This is a very minor optimization for a very uncommon case (I believe), so we should in fact remove that condition completely. What do you think?

Reply:
agreed, the change I added should cover that case as well, so the top if is no longer necessary.
but I also don't quite understand the next if:
`        if not (
            self._bb_valid and all(ref._bb_valid for ref in self.get_dependencies(True))
        ):`
is this necessary, or can this also be removed?

Reply:
oh I think I understand now... you use this to cache the bbox value? and I assume you set self._bb_valid to False in any place which adds or removes any polygons/refs/etc? so you know if you need to update it?

Reply:
i removed the top `if` and made the code a bit more readable. should be good to go

Reply:
sorry, got distracted with the bottom comments and forgot about the first one... _now_ it should be good to go. you'll notice that a few other lines had to be shuffled around in order to properly cache the `_bounding_box = None` case.